### PR TITLE
Added --skip-benchmarks command-line option.

### PR DESCRIPTION
**Pull request by danifeist**
This PR adds the feature requested by https://github.com/catchorg/Catch2/issues/2392 "Command line option to skip benchmarks" by introducing the command-line option "--skip-benchmarks" so that Catch runs no benchmarks when this option is set. I'd be glad about feedback.

### CODE REVIEW

File: docs/command-line.md [hunk 1]
```diff
@@ -370,6 +372,12 @@ See [The LibIdentify repo for more information and examples](https://github.com/
 Will cause the executable to print a message and wait until the return/ enter key is pressed before continuing -
 either before running any tests, after running all tests - or both, depending on the argument.
 
+<a id="skip-benchmarks"></a>
+## Skip all benchmarks
+<pre>--skip-benchmarks</pre>
+
```
Comment (horenmar) [inline docs/command-line.md:378]: This should have a version placeholder (see contributing.md for how it looks).

File: docs/command-line.md [hunk 2]
```diff
@@ -370,6 +372,12 @@ See [The LibIdentify repo for more information and examples](https://github.com/
...
+When this option is set, all benchmarks of each and every Catch test case will be skipped so that it appears as if there were no benchmark defined.
```
Comment (horenmar) [inline docs/command-line.md]: How about

```suggestion
This flag tells Catch2 to skip running all benchmarks. Benchmarks in this
case mean code blocks in `BENCHMARK` and `BENCHMARK_ADVANCED` macros, not
test cases with the `[!benchmark]` tag.
```

I think this is clearer. (Also new docs use Catch2 and line length of 80, to play nicer with diffs)

File: src/catch2/benchmark/catch_benchmark.hpp [hunk 3]
```diff
@@ -96,7 +96,10 @@ namespace Catch {
             template <typename Fun, std::enable_if_t<!Detail::is_related<Fun, Benchmark>::value, int> = 0>
                 Benchmark & operator=(Fun func) {
                 fun = Detail::BenchmarkFunction(func);
-                run();
+                auto const* cfg = getCurrentContext().getConfig();
+                if (!cfg->skipBenchmarks()) {
+                    run();
+                }
```
Comment (horenmar) [inline src/catch2/benchmark/catch_benchmark.hpp:102]: I think this should happen before we create the function wrapper.

**Review by horenmar — CHANGES_REQUESTED** [review]
This also needs tests.

It should be enough to pick some TEST_CASE that's already in the SelfTest project and has benchmarks, and then add new CTest test that invokes `SelfTest` with `--skip-benchmarks` and test spec for that one test. The test then checks that the benchmark was skipped by looking for the benchmark output.

Or add a new TEST_CASE just for this, that way is more resilient to later changes.

File: src/catch2/benchmark/catch_benchmark.hpp (hunk 3 above)
Comment (danifeist) [reply src/catch2/benchmark/catch_benchmark.hpp:102]: you'd place `fun = Detail::BenchmarkFunction(func);` inside the if-clause, did i got that right?

Comment (horenmar) [reply src/catch2/benchmark/catch_benchmark.hpp:102]: Yeah. There is no reason to make the allocation if it won't be used afterwards.

File: tests/CMakeLists.txt [hunk 4]
```diff
@@ -359,6 +359,18 @@ add_test(NAME CheckConvenienceHeaders
...
+  COMMAND
+    $<TARGET_FILE:SelfTest> "Skip benchmark macros"
+      --reporter console
+      --skip-benchmarks
+)
+set_tests_properties("Benchmarking::SkipBenchmarkMacros"
+  PROPERTIES
+    PASS_REGULAR_EXPRESSION "All tests passed \\(2 assertions in 1 test case\\)"
+    FAIL_REGULAR_EXPRESSION "benchmark name"
+)
```
Comment (horenmar) [inline tests/CMakeLists.txt:372]: I tightened up the tests a bit. This way the tests fail if either the assertions disappear for some reason, or if the "benchmark begins" event gets sent (even if it isn't executed afterwards).

**Comment by horenmar** [comment]
Thanks.

**Pull request merged**